src/StreamHandler.h

```diff
@@ -111,6 +119,23 @@ class StreamMetricsHandler : public StreamHandler
         _metrics->configure_groups(&_groups);
     }
 
+    void validate_configs(const ConfigsDefType &config_defs)
+    {
+        auto all_configs = get_all_keys();
+        for (const auto &config : all_configs) {
+            if (std::any_of(_window_config_defs.begin(), _window_config_defs.end(), [config](const auto &def) { return config == def; })) {
+                continue;
+            } else if (std::any_of(config_defs.begin(), config_defs.end(), [config](const auto &def) { return config == def; })) {
+                continue;
+            } else if (config == "enable") {
+                continue;
+            } else if (config == "disable") {
+                continue;
+            }
+            throw StreamHandlerException(fmt::format("{} is an invalid/unsupported config or filter. The valid configs/filters are: {}", config, fmt::join(config_defs, ", ")));
```

Comment:
@weyrick Should I return the window configs like `deep_sample_rate` here as well?

Reply:
hm, i guess so